### DISCUSSION TITLE
네이버 지도 graphql JSON 관련 응답값 변화 (지도 새로고침 방식 사용불가)

### DISCUSSION THREAD

**Discussion by SJang1** [changdoc/naver-vaccine-macro#84] [Show and tell]
안녕하세요.
본 프로젝트에 영감을 얻어서 그동안 간단하게 파이썬으로 현재 하시려는 프로젝트와 같이 지도 화면에서 응답이 오는 graphql JSON의 응답으로 잔여백신이 있는 경우 잔여백신 예약 페이지를 띄워서, 본 매크로를 통해 예약을 시도하고자 하고 있었습니다.

그러나, 매크로를 이용해서 잔여백신을 검색 시 **네이버에서 잔여백신이 존재하는 딱 그 타이밍에 모든 병원의 모든 정보를 보내주지 않는, _모든 items가 없는 상태로_** 응답이 오는 것을 확인하여 Discuss 남깁니다. (이것이 네이버가 새로고침 매크로를 차단한 방법인 것으로 보입니다. 카카오의 경우 관련된 제한이 아직은 없습니다.)

해당 문제로 인해 이전부터 네이버가 아닌 카카오의 지도 API를 이용해서 잔여백신 존재 시 네이버 예약 페이지로 이동하는 방식을 이용했으나, 네이버 잔여백신 예약 시 sid값이 필요하게 됨에 따라 해당 방식도 더이상 이용할 수 없게 되었습니다.

따라서, 완전히 네이버를 무시하고 카카오 잔여백신을 통해 예약을 하는 방법을 찾아서 해당 방법으로 문제없이 예약을 진행하고 있습니다. 제가 보기에는 현재 방법 외에 네이버는 방법이 딱히 없을 것 같습니다.

참고하시라고 관련된 제 프로젝트 링크 남겨드립니다 : [SJang1/korea-covid-19-remaining-vaccine-macro](https://github.com/SJang1/korea-covid-19-remaining-vaccine-macro)

**Comment by changdoc** [comment]
오호 더 확실하고 빠르게 예약 가능한 형태로 보이네요!

**Comment by changdoc** [comment]
@SJang1 
지도 새로고침 방식으로 제작하여 테스트를 진행 중이였는데 모든 items가 없는 상태로 응답 <-- 이 부분이 백신 발생하는 타이밍에만 오는거면 활용이 힘들겠네요.

해서 저도 카카오 맵에서 잔여 백신 조회 api를 매크로 코드에서 주기적으로 체크하여 발생할 경우 네이버 예약 신청 페이지로 이동 되게 기능을 추가 후 확인 중에 있습니다.

네이버 예약 페이지로 이동시 필요한 데이터는 기존에 병원 목록을 만들때 사용한 데이터에서 참조를 하는 방식입니다.

카카오 api결과에서 orgCode를 가지고 네이버 graphql 결과(병원목록)에서 해당 병원의 데이터에 있는 sid를 조합하여 예약 신청 페이지 이동 -> 기존 매크로 신청 동작 으로 이동하는 방식인데..

네이버는 예약 신청 페이지에 접근해서 key를 발급받아서 신청을 진행해야해서 신청 페이지를 반드시 거쳐야하는 구조라
파이썬에서 바로 예약 신청을 진행하는 페이지를 바로 열어버리는 방식이 딜레이가 적어 확률이 더 높을것 같네요.

**Reply by SJang1** [reply]
> 지도 새로고침 방식으로 제작하여 테스트를 진행 중이였는데 모든 items가 없는 상태로 응답 <-- 이 부분이 백신 발생하는 타이밍에만 오는거면 활용이 힘들겠네요.

네.. 정확한 rate limit을 걸어서 잔여백신 개수 표시 안되게 하는 타이밍이 언제인지는 테스트를 못해봤는데 어저께 네이버랑 카카오를 동시에 돌리며 비교해보았는데 정확히 잔여백신이 발생했을 때에만 모든 items가 없는 상태로 응답하더라고요..

그래도 잔여백신 발생 직후에는 DB딜레이가 있어서 너무 빠르게 예약을 시도하면 예약시도를 실패하는걸로 보여서, 어느 정도 딜레이가 있는 지금 개발하고 계신 방식이 더 가능성이 낮다고만 할 수는 없을거같습니다.

**Reply by MKK425** [reply]
해당 프로젝트 들어가서 설치하려는데 지도 이동하면 가끔 지원하지 않는 환경입니다?라고 뜨는데 작동은 될까요?
그리고 개발자도구를 열어 x,y 좌표 위치를 확인하라고 하였는데 해당부분을 어떻게 확인하는지 모르겠습니다.

**Reply by SJang1** [reply]
시작에 앞서 맞는 프로젝트에 질문을 올려주시면 감사드리겠습니다.

> 해당 프로젝트 들어가서 설치하려는데 지도 이동하면 가끔 지원하지 않는 환경입니다?라고 뜨는데 작동은 될까요?

https://github.com/SJang1/korea-covid-19-remaining-vaccine-macro/discussions/7#discussioncomment-1040629

> 그리고 개발자도구를 열어 x,y 좌표 위치를 확인하라고 하였는데 해당부분을 어떻게 확인하는지 모르겠습니다.

https://github.com/SJang1/korea-covid-19-remaining-vaccine-macro/discussions/2
